src/tests/baseservices/exceptions/simple/ParallelCrash.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+// Runtime stability in the presence of concurrent fatal errors
+
+public class ParallelCrash
+{
+    private const int ThreadCount = 10;
+
+    private volatile static int _runningThreads;
+
+    public static int Main()
+    {
+        Thread[] threadMap = new Thread[ThreadCount];
+        for (int threadIndex = 0; threadIndex < ThreadCount; threadIndex++)
+        {
+            Thread thread = new Thread(new ThreadStart(CrashInParallel));
+            threadMap[threadIndex] = thread;
+            thread.Start();
+        }
+        for (;;)
+        {
+            Thread.Sleep(1000);
+        }
+        return 0;
+    }
+
+    private static void CrashInParallel()
+    {
+        int threadIndex = ++_runningThreads;
+        string failFastMessage = string.Format("Parallel crash in thread {0}!\n", threadIndex);
+        while (_runningThreads != ThreadCount)
+        {
+        }
+        // Now all the worker threads should be running, fire!
+        Environment.FailFast(failFastMessage);
```

Comment:
It would be good to also have another test to exercise the case when the failfast comes from several secondary threads.

Reply:
Well, as of the 3rd commit that's exactly what the test does - it hangs on the main thread and crashes all the worker threads in parallel. I have added two more variations - one to just crash the main thread and another to crash both the main thread and the worker threads. Please let me know if you believe we need to further improve the test coverage.